babel/messages/frontend.py

```diff
@@ -295,13 +294,18 @@ def finalize_options(self):
                     'input-dirs and input-paths are mutually exclusive'
                 )
 
-        if self.no_default_keywords and not self.keywords:
-            raise DistutilsOptionError('you must specify new keywords if you '
-                                       'disable the default ones')
         if self.no_default_keywords:
-            self._keywords = {}
-        if self.keywords:
-            self._keywords.update(parse_keywords(self.keywords.split()))
+            keywords = {}
+        else:
+            keywords = DEFAULT_KEYWORDS.copy()
+
+        for kwarg in (self.keywords or ()):
+            keywords.update(parse_keywords(kwarg.split()))
+
+        self.keywords = keywords
+
+        if not self.keywords:
+            raise DistutilsOptionError('no keywords specified')
```

Comment:
I think the old error message ('you must specify new keywords if you disable the default ones') was more descriptive, given that's the only time this will happen.